types/jquery/ts3.9/misc.deferred.d.ts

```diff
@@ -0,0 +1,1125 @@
+            RRD, RJD, RND>;
+
+        // #endregion
+
+        // region then
+        // #region then
+
+        /**
+         * Add handlers to be called when the Deferred object is resolved, rejected, or still in progress.
+         * @param doneFilter An optional function that is called when the Deferred is resolved.
+         * @param failFilter An optional function that is called when the Deferred is rejected.
+         * @param progressFilter An optional function that is called when progress notifications are sent to the Deferred.
+         * @see \`{@link https://api.jquery.com/deferred.then/ }\`
+         * @since 1.8
+         * @example ​ ````Since the jQuery.get method returns a jqXHR object, which is derived from a Deferred object, we can attach handlers using the .then method.
+```javascript
+$.get( "test.php" ).then(
+  function() {
+    alert( "$.get succeeded" );
+  }, function() {
+    alert( "$.get failed!" );
+  }
+);
+```
+         * @example ​ ````Filter the resolve value:
+```html
+<!doctype html>
+<html lang="en">
+<head>
+  <meta charset="utf-8">
+  <title>deferred.then demo</title>
+  <script src="https://code.jquery.com/jquery-3.3.1.js"></script>
+</head>
+<body>
+​
+<button>Filter Resolve</button>
+<p></p>
+​
+<script>
+var filterResolve = function() {
+  var defer = $.Deferred(),
+    filtered = defer.then(function( value ) {
+      return value * 2;
+    });
+​
+  defer.resolve( 5 );
+  filtered.done(function( value ) {
+    $( "p" ).html( "Value is ( 2*5 = ) 10: " + value );
+  });
+};
+​
+$( "button" ).on( "click", filterResolve );
+</script>
+​
+</body>
+</html>
+```
+         * @example ​ ````Filter reject value:
+```javascript
+var defer = $.Deferred(),
+  filtered = defer.then( null, function( value ) {
+    return value * 3;
+  });
+​
+defer.reject( 6 );
+filtered.fail(function( value ) {
+  alert( "Value is ( 3*6 = ) 18: " + value );
+});
+```
+         * @example ​ ````Chain tasks:
+```javascript
+var request = $.ajax( url, { dataType: "json" } ),
+  chained = request.then(function( data ) {
+    return $.ajax( url2, { data: { user: data.userId } } );
+  });
+​
+chained.done(function( data ) {
+  // data retrieved from url2 as provided by the first request
+});
+```
+         */
+        then<ARD = never, AJD = never, AND = never,
+            BRD = never, BJD = never, BND = never,
+            CRD = never, CJD = never, CND = never,
+            RRD = never, RJD = never, RND = never,
+            ARF = never, AJF = never, ANF = never,
+            BRF = never, BJF = never, BNF = never,
+            CRF = never, CJF = never, CNF = never,
+            RRF = never, RJF = never, RNF = never,
+            ARP = never, AJP = never, ANP = never,
+            BRP = never, BJP = never, BNP = never,
+            CRP = never, CJP = never, CNP = never,
+            RRP = never, RJP = never, RNP = never>(
+                doneFilter: (t: awaited TR, u: UR, v: VR, ...s: SR[]) => PromiseBase<ARD, AJD, AND,
```

Comment:
```
../jquery/ts3.9/misc.deferred.d.ts(721,30): error TS2502: 't' is referenced directly or indirectly in its own type annotation.
```

Looks like a bug? The same appears in the other uses of `t: awaited TR`.

Reply:
I'm looking into what's causing this.